raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2261,6 +2260,11 @@ private Long append(int epoch, List<T> records, boolean isAtomic) {
         return shutdownComplete;
     }
 
+    @Override
+    public void resign(int epoch) {
+        throw new UnsupportedOperationException();
```

Comment:
This needs to be supported, because the controller will resign if it detects certain bugs.

Reply:
Yes. We have an issue for this: https://issues.apache.org/jira/browse/KAFKA-12631

Reply:
For some reason, I thought this was already implemented. But you're correct, it is not currently implemented. Maybe we should have this for 3.0, since controller bugs do happen sometimes...